Core/src/org/sleuthkit/autopsy/corecomponents/DataContentViewerArtifact.java

```diff
@@ -344,25 +354,49 @@ public boolean isSupported(Node node) {
         }
         return false;
     }
-
+    
+      
     @Override
     public int isPreferred(Node node) {
+        // get the artifact from the lookup
         BlackboardArtifact artifact = node.getLookup().lookup(BlackboardArtifact.class);
-        // low priority if node doesn't have an artifact (meaning it was found from normal directory
-        // browsing, or if the artifact is something that means the user really wants to see the original
-        // file and not more details about the artifact
-        if ((artifact == null)
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_HASHSET_HIT.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_INTERESTING_FILE_HIT.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_OBJECT_DETECTED.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_METADATA_EXIF.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_EXT_MISMATCH_DETECTED.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_WEB_DOWNLOAD.getTypeID())
-                || (artifact.getArtifactTypeID() == ARTIFACT_TYPE.TSK_WEB_CACHE.getTypeID())) {
-            return 3;
-        } else {
-            return 6;
+        
+        // if there is an artifact, get the type
+        BlackboardArtifact.Type artifactType = null;
+        if (artifact != null) {
+            try {
+                artifactType = artifact.getType();
+            } catch (TskCoreException ex) {
+                logger.log(Level.WARNING, 
+                        String.format("There was an error getting the artifact type for artifact with id: %d", artifact.getId()), 
+                        ex);
+            }
+        }
+        
+        // if web download or web cache, less preferred.
+        if (artifactType != null && 
+                (artifactType.getTypeID() == BlackboardArtifact.Type.TSK_WEB_DOWNLOAD.getTypeID() || 
+                artifactType.getTypeID() == BlackboardArtifact.Type.TSK_WEB_CACHE.getTypeID())) {
+            
+            return LESS_PREFERRED;
+        }
+             
+        // if there is a type, get the category 
+        Category category = artifactType == null ? null : artifactType.getCategory();
+        
+        // if no category, treat as less preferred.
+        if (category == null) {
+            return LESS_PREFERRED;
+        }
+        
+        switch (category) {
+            // data artifacts should be more preferred
+            case DATA_ARTIFACT:
+                return MORE_PREFERRED;
+            // everything else is less preferred
+            case ANALYSIS_RESULT:
+            default:
+                return LESS_PREFERRED;
```

Comment:
I am finding this logic a bit hard to follow with some unnecessary null checks and setting of local variables to null to act as flow of control. Consequently, it's not clear to me that we are avoiding all NPEs. If there is no artifact in the lookup, we need not proceed any further.  BlackboardArtifact.getType() will always return a value or an exception will be thrown, so we don't need to null check that. The Type will always have a category, so we don't need to null check that. In any case, let's try to make this easier to read, please. Also, we should comment on why we are treating web download and cache artifacts differently. It's because for these artifacts, the important thing is the content that was downloaded or cached, not the artifact itself.